components/brave_ads/browser/component_updater/component_util.cc

```diff
@@ -10,8 +10,8 @@
 namespace brave_ads {
 
 absl::optional<ComponentInfo> GetComponentInfo(const std::string& id) {
-  const auto iter = components.find(id);
-  if (iter == components.end()) {
+  const auto iter = kComponents.find(id);
```

Comment:
holy moly. the huge `kComponents` vector declared in `components/brave_ads/browser/component_updater/component_util.h` is crazy! please move it into `.cc` file as constexpr or as `base::NoDestructor` accessible via `GetComponent()` global function.